### PR TITLE
Reconfigure `flake8` to fail build if it finds any errors

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR removes the second `flake8` run that happens when checking code before approving it to merge into `SC_redesign`. 

It also fixes the handful of linter errors that were present in `SC_redesign` when this reconfiguration was done.

### CODE REVIEW

File: .github/workflows/lint_flake8.yml [hunk 1]
```diff
@@ -27,7 +27,5 @@ jobs:
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
     - name: Lint with flake8
       run: |
-        # stop the build if there are Python syntax errors or undefined names
-        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (morrowcj) [inline .github/workflows/lint_flake8.yml:31]: What are we losing by removing the `--select` flag? I don't understand what these values indicate.

**Review by morrowcj — APPROVED** [review]

File: .github/workflows/lint_flake8.yml (hunk 1 above)
Comment (Isgotkowitz) [reply .github/workflows/lint_flake8.yml:31]: Those are different error codes. It tells flake8 to stop the build **only** if it finds errors listed in the `select` flag. Check out [error codes](https://flake8.pycqa.org/en/latest/user/error-codes.html) for more info

**Review by matthew7838 — APPROVED** [review]
Looks like a good change to linting. One question, are the changes to other files addressing the warning issues brought up by the updated lint?

**Comment by Isgotkowitz** [comment]
> Looks like a good change to linting. One question, are the changes to other files addressing the warning issues brought up by the updated lint?

Yes and no. The issues were raised by both versions of the linter, but in the old version it was set so that they were not fatal and you wouldn't know they had been found unless you looked through the giant list of warnings raised by the linter after running over the entirety of the codebase. Which is I am very gung-ho on changing the linting config, the old workflow IMO has very bad usability.

With the new linting workflow when there are formatting errors, it will (fingers crossed) fail the linting check, list where all the errors were found, and won't let the PR be merged until they get fixed.

**Pull request merged**